### PR TITLE
Feature/context improvement

### PR DESCRIPTION
**Pull request by bjornandre**

### CODE REVIEW

File: src/context/ContextProvider.js [hunk 1]
```diff
@@ -26,11 +28,21 @@ export class ContextProvider extends Component {
...
     }
-}
+  }
+
+  getLocalizedText (state) {
+    return (codeTextObj) => {
+      let text = codeTextObj.find(name => name.languageCode === state.languageCode) || codeTextObj[0]
+      return text === undefined ? null : text.languageText
+    }
+  }
```
Comment (takvamborgen) [inline src/context/ContextProvider.js]: Add test for this

File: src/context/ContextProvider.js [hunk 2]
```diff
@@ -26,11 +28,21 @@ export class ContextProvider extends Component {
...
     datasetService: Properties.mock.datasetService === true ? new DatasetServiceMock(): new DatasetServiceImpl(),
-    notebookService: Properties.mock.notebookService === true ? new NotebookServiceMock(): new NotebookServiceImpl(),
-    setLanguage: (language) => {
-      this.setState({ languageCode: language })
+    notebookService: Properties.mock.notebookService === true ? new NotebookServiceMock(): new NotebookServiceImpl()
+  }
+
+  setLanguage (ref) {
+    return (language) => {
+      ref.setState({ languageCode: language })
```
Comment (takvamborgen) [inline src/context/ContextProvider.js]: Add test for this

File: src/context/ContextProvider.js (hunk 1 above)
Comment (bjornandre) [reply src/context/ContextProvider.js]: Fixed

File: src/context/ContextProvider.js (hunk 2 above)
Comment (bjornandre) [reply src/context/ContextProvider.js]: Fixed

**Pull request merged**